fix(category_theory/types): rename lemma `ulift_functor.map`

urkud opened the pull request:
This way, one can use `ulift_functor.map` in the sense `functor.map ulift_functor`.

TO CONTRIBUTORS:

Make sure you have:

  * [X] reviewed and applied the coding style: [coding](https://github.com/leanprover/mathlib/blob/master/docs/contribute/style.md), [naming](https://github.com/leanprover/mathlib/blob/master/docs/contribute/naming.md)
  * [ ] for tactics:
     * [ ] added or adapted documentation in [tactics.md](https://github.com/leanprover/mathlib/blob/master/docs/tactics.md)
     * [ ] write an example of use of the new feature in [tactics.lean](https://github.com/leanprover/mathlib/blob/master/tests/tactics.lean)
  * [X] make sure definitions and lemmas are put in the right files
  * [X] make sure definitions and lemmas are not redundant

If this PR is related to a discussion on Zulip, please include a link in the discussion.

For reviewers: [code review check list](https://github.com/leanprover/mathlib/blob/master/docs/contribute/code-review.md)

urkud committed:
fix(category_theory/types): avoid shadowing `ulift_functor.map` by a lemma

Now we can use `ulift_functor.map` in the sense `functor.map ulift_functor`.

kim-em reviewed (dismissed):
Good idea to fix this.

I think the canonical name would be `ulift_functor_map`.

urkud committed:
`ulift_functor.map_spec` → `ulift_functor_map`

as suggested by @semorrison in https://github.com/leanprover-community/mathlib/pull/1133#pullrequestreview-250179914

mergify[bot] committed:
Merge branch 'master' into ulift_functor.map
